internal/http/client.go

```diff
@@ -0,0 +1,33 @@
+package http
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+)
+
+type Client struct {
+	client *http.Client
+}
+
+func NewHTTPClient(tlsConfig *tls.Config) (*Client, error) {
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsConfig,
+		},
+	}
+
+	return &Client{
+		client: client,
+	}, nil
+}
+
+func (c *Client) Get(url string) (*http.Response, error) {
```

Comment:
Just note. I'm working on the code that reads consumer/ent. certs/key and set up connections, keep connections, allow to do all HTTP requests (not only Get), debug print of HTTP requests, etc. Thus we will be able to share this code in the future as Go package.